juju/juju
# Do Not Track Active Generation in State

## Description of change

- Modifies `switch-generation` command to check whether there is an active next generation when attempting to set the generation to "next".
- Removes methods relating to generation switching from API clients and servers.
- Removes the `Active` member from `Generation` in the state package along with logic relating to it.
- Replaces transaction assertions that deal with the `Active` value, with logic to ensure that a generation is not yet completed (which `Active` was an effective proxy for).

## QA steps

- `export JUJU_DEV_FEATURE_FLAGS=generations`.
- Bootstrap.
- `juju deploy redis -n 2`
- `juju add-generation`.
- `juju advance-generation redis/0`.
- `juju cancel-generation` should fail.
- `juju switch-generation current`.
- `juju advance-generation redis` (all remaining will auto-complete generation).
- `juju switch-generation next` should fail due to no "next" generation being found.

## Documentation changes

None.

## Bug reference

N/A
